src/Serializer/Filter/GroupFilter.php

```diff
@@ -24,11 +24,13 @@
 {
     private $overrideDefaultGroups;
     private $parameterName;
+    private $whitelist;
 
-    public function __construct(string $parameterName = 'groups', bool $overrideDefaultGroups = false)
+    public function __construct(string $parameterName = 'groups', bool $overrideDefaultGroups = false, array $whitelist = [])
```

Comment:
I suggest to use `array $whitelist = null` and to check if the value if not `null` instead. It will be more intuitive, we have a special value for a special case.

Reply:
Sure I'll update tonight